modules/api/php/endpoints/candidate/visit/instrument/flags.class.inc

```diff
@@ -127,6 +128,108 @@ class Flags extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PUT request.
+     * Creates or updates all statuses for a given instrument with new values.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request) : ResponseInterface
+    {
+        // TODO :: Check permissions. How??
+        $this->_instrumentStatus = new \NDB_BVL_InstrumentStatus();
+        $this->_instrumentStatus->select($this->_instrument->commentID);
+
+        $data = json_decode((string) $request->getBody(), true);
+
+        try {
+
+            $requiredfields = [
+                'Data_entry',
+                'Administration',
+                'Validity',
+            ];
+
+            $diff = array_diff($requiredfields, array_keys($data['Flags']));
+            if (!empty($diff)) {
+                return new \LORIS\Http\Response\JSON\BadRequest(
+                    'Field(s) missing in Flags: ' . implode(', ', $diff)
+                );
+            }
+            $this->_instrumentStatus->setDataEntryStatus(
+                $data['Flags']['Data_entry']
+            );
+            $this->_instrumentStatus->setAdministrationStatus(
+                $data['Flags']['Administration']
+            );
+            $this->_instrumentStatus->setValidityStatus(
+                $data['Flags']['Validity']
+            );
+
+        } catch (\Throwable $e) {
+            error_log($e->getMessage());
+            return new \LORIS\Http\Response\JSON\InternalServerError();
+        }
+        return (new \LORIS\Http\Response())
+            ->withStatus(204);
+    }
+
+    /**
+     * Handles a PATCH request.
+     * Creates or updates some or all statuses for a given instrument. If all
+     * statuses are requested to be changed, this method is equivalent to handlePUT.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePATCH(ServerRequestInterface $request): ResponseInterface
+    {
+        // TODO :: Check permissions. How??
```

Comment:
You can check:

1. Does the user have the `data_entry` permission?
2. The value of `$this->_instrument->_hasAccess($user)`

(same with `PUT`)

Reply:
these will be resolved in #6780. `data_entry` is already checked